[Security Solution] Improves Cypress execution time (#127558)
elastic/kibana: x-pack/plugins/security_solution/cypress/README.md

## Development Best Practices
Below you will a set of best practices that should be followed when writing Cypress tests.

### Write easy to maintain tests
Consider to extract all the elements you need to interact with to the `screens` folder. In this way in case the locator changes, we just need to update the value in one place.

### Write easy to read tests
Consider to extract all the tasks a user should perfom into the `tasks` folder. In this way is going to be easier to undertsand what are we trying to mimic from the user perspective. Also in case there is change on the way the user has to perform the action, we 

> just need to update the value in one place.

Reviewer: 💅 This seems to have jumped to the next line. Consider removing the line break.

> ### Make sure your test fail
> Before open a PR with the new test, please make sure that the test fail. If you never see your test fail you don’t know if your test is actually testing the right thing, or testing anything at all.

Reviewer: ```suggestion
### Make sure your test fails
Before open a PR with a new test, please first make sure that the test fails. If you never see your test fail you don’t know if your test is actually testing the right thing, or testing anything at all.
```

- Group the tests that are similar in different contexts.
- For every context login only once, clean the state between tests if needed without re-loading the page.
- All tests in a spec file must be order-independent.

> - Clean up the state and data just when needed using `cleanKibana` function.  The execution of it is expensive of time, so consider if you really need to clean the data before the execution. I.e: If you are just checking that a modal can be opened, you may not need to clean the data.

Reviewer: ```suggestion
- Clean up the state and data just when needed using `cleanKibana` function.  Executing this function takes a lot of time, so consider if you really need to clean the data before the execution. I.e: If you are just checking that a modal can be opened, you may not need to clean the data.
```
